File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] height) {

        int l = 0;
        int r = height.length - 1;

        int maxTotal = 0;

        while (l < r) {
            int water = Math.min(height[l], height[r]) * (r - l);
            if (water > maxTotal) maxTotal = water;
            if (height[l] < height[r]) {
                l ++;
            }
            else {
                r --;
            }
        }

        return maxTotal;
    }
}

File: Data Structures & Algorithms/interval-list-intersections/submission-0.cs
public class Solution {
    public int[][] IntervalIntersection(int[][] firstList, int[][] secondList) {
        List<int[]> intervals = new List<int[]>();

        var firstPointer = 0;
        var secondPointer = 0;

        while (firstPointer < firstList.Length && secondPointer < secondList.Length) {
            var fStart = firstList[firstPointer][0];
            var fEnd = firstList[firstPointer][1];

            var sStart = secondList[secondPointer][0];
            var sEnd = secondList[secondPointer][1];

            var start = Math.Max(fStart, sStart);
            var end = Math.Min(fEnd, sEnd);

            if (start <= end) {
                intervals.Add([start, end]);
            }

            if (fEnd < sEnd) firstPointer ++;
            else secondPointer ++;
        }

        return intervals.ToArray();
    }
}